Mu2eG4/src/constructDiskCalorimeter.cc

```diff
@@ -289,48 +288,117 @@ namespace mu2e {
        reg.add(rotFPPipe);
        
        G4Torus*         coolFP     = new G4Torus("caloCoolFP",FPCoolPipeRadius-FPCoolPipeThickness, FPCoolPipeRadius, FPCoolPipeTorRadius, angMax, CLHEP::twopi-2.0*angMax);
-       G4LogicalVolume* coolFPLog  = caloBuildLogical(coolFP, pipeMaterial, "caloCoolFPLog_"+std::to_string(idisk),isPipeVisible,G4Color::Red(),isPipeSolid,0);
+       G4LogicalVolume* coolFPLog  = caloBuildLogical(coolFP, pipeMaterial, "caloCoolFPLog",isPipeVisible,G4Color::Red(),isPipeSolid,0);
        pv = new G4PVPlacement(rotFPPipe,G4ThreeVector(0.0,0.0,ZposPipe), coolFPLog, "caloCoolFPPV", frontPlateLog, false, 0, false);
        doSurfaceCheck && checkForOverlaps( pv, config, verbosityLevel>0);                
        
 
        //pipes nside foam
        G4RotationMatrix* rotPipe1 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipe1->rotateZ(CLHEP::pi/2.0);
+       //rotPipe1->rotateZ(CLHEP::pi/2.0);
+       rotPipe1->rotateZ(0.0 * CLHEP::pi);
        G4RotationMatrix* rotPipe2 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipe2->rotateZ(1.5*CLHEP::pi);
-       G4RotationMatrix* rotPipeFlat = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
-       rotPipeFlat->rotateX(CLHEP::pi/2.0);
+       rotPipe2->rotateZ(1.0 * CLHEP::pi); 
+       G4RotationMatrix* rotPipe3 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
+       rotPipe3->rotateY(1.0 * CLHEP::pi);
+       G4RotationMatrix* rotPipe4 = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
+       rotPipe4->rotateX(1.0 * CLHEP::pi);
+       //G4RotationMatrix* rotPipeFlat = new G4RotationMatrix(CLHEP::HepRotation::IDENTITY);
+       //rotPipeFlat->rotateX(CLHEP::pi/2.0);
        reg.add(rotPipe1);
        reg.add(rotPipe2);
-       reg.add(rotPipeFlat);
+       reg.add(rotPipe3);
+       reg.add(rotPipe4);
+       //reg.add(rotPipeFlat);
```

Comment:
It will be easier to maintain if reg.add(a) are placed just after new a(...) statements

Reply:
Actually, one liners like: T* a = reg.add(new (a(..)) are probably most optimal